walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,54 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    ArrayList<TimeRange> openTimes = new ArrayList<TimeRange>();
+
+    // Edge case when a meeting is longer than a full day.
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()){
```

Comment:
what happens when there are no events? Did you consider this edge case?

Reply:
`// If there is time for the meeting after all of the events, add that time.
    if (dayStart + request.getDuration() <= TimeRange.END_OF_DAY) {
      openTimes.add(TimeRange.fromStartEnd(dayStart, TimeRange.END_OF_DAY, true));
    }`
This if statement at the end here also accounts for the scenario if there are no events in a day.